### PR TITLE
Homework Podstawy C++ password-check

### PR DESCRIPTION
**Pull request by opiotrowsky**
Authors: @opiotrowsky @PawelZydek

### CODE REVIEW

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @opiotrowsky

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @PawelZydek

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @opiotrowsky

**Comment by ziobron** [comment]
⏱ 3 XP extra for @opiotrowsky for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @PawelZydek

**Comment by ziobron** [comment]
⏱ 3 XP extra for @PawelZydek for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @opiotrowsky

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,100 @@
...
+        break;
+    }
+    case 4: {
+        return "Password needs to have at least one uppercase letter";
+        break;
+    }
+    case 5: {
+        return "Passwords do not match";
+        break;
+    }
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:35]: Better use enum, thats why why named variable!

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,100 @@
...
+}
+
+bool doPasswordsMatch(std::string password, std::string repeatedPassword) {
+    if (password == repeatedPassword) {
+        return true;
+    }
+    else {
+        return false;
+    }
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:45]: Just `return lhs == rhs`

File: homework/password-check/validation.cpp [hunk 3]
```diff
@@ -1,2 +1,100 @@
...
+//    case 3: {
+//        return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+//        break;
+//    }
+//    case 4: {
+//        return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
+//        break;
+//    }
+//    }
+//}
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:74]: remove commented code

File: homework/password-check/validation.cpp [hunk 4]
```diff
@@ -1,2 +1,100 @@
...
+    if (password.length() < 9) {
+        return ErrorCode::PasswordNeedsAtLeastNineCharacters;
+    } else if (std::none_of(password.begin(), password.end(), isdigit)) {
+        return ErrorCode::PasswordNeedsAtLeastOneNumber;
+    } else if (std::none_of(password.begin(), password.end(), ispunct)) {
+        return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+    } else if (std::none_of(password.begin(), password.end(), isupper)) {
+        return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
+    } else {
+        return ErrorCode::Ok;
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:88]: When you use return inside `if` you don't need `if else` or `else`. Just write `if (sth) { return blabla; }`

File: homework/password-check/validation.cpp [hunk 5]
```diff
@@ -1,2 +1,100 @@
...
+//        break;
+//    }
+//    case 4: {
+//        return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
+//        break;
+//    }
+//    }
+//}
+
+ErrorCode checkPasswordRules(std::string password) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:76]: Pass by const refrence. Try always avoid copy.

File: homework/password-check/validation.cpp [hunk 6]
```diff
@@ -1,2 +1,100 @@
...
+    } else if (std::none_of(password.begin(), password.end(), ispunct)) {
+        return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+    } else if (std::none_of(password.begin(), password.end(), isupper)) {
+        return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
+    } else {
+        return ErrorCode::Ok;
+    }   
+}
+
+ErrorCode checkPassword(std::string password, std::string repeatedPassword) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:92]: Pass by const refrence. Try always avoid copy.

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**